### PR TITLE
Update minimum PyMC dependency 

### PR DESCRIPTION
**Pull request by cetagostini**
----
:books: Documentation preview :books:: https://pymc-marketing--385.org.readthedocs.build/en/385/

### CODE REVIEW

**Comment by cetagostini** [comment]
@ricardoV94

**Review by twiecki — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
     branches: [main]
 
 env:
-  OLDEST_PYMC_VERSION: "5.7.1"
+  OLDEST_PYMC_VERSION: "5.8.2"
```
Comment (twiecki) [inline .github/workflows/ci.yml:10]: Is htis actually required?

Comment (cetagostini) [reply .github/workflows/ci.yml:10]: The comment on the `.toml` file mentioned the following:

1. "NOTE: Keep minimum Pymc version in sync with ci.yml `OLDEST_PYMC_VERSION`"

As well, @ricardoV94 mentioned I should modify `pyproject` and the `ci` files.

File: pyproject.toml [hunk 2]
```diff
@@ -14,10 +14,10 @@ maintainers = [{ name = "PyMC Labs", email = "info@pymc-labs.io" }]
 dependencies = [
     "arviz>=0.13.0",
     "matplotlib>=3.5.1",
-    "numpy>=1.17",
+    "numpy>=1.17,<1.25",
```
Comment (ricardoV94) [inline pyproject.toml:17]: We shouldn't have to pin numpy and PyTensor does it already

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ricardoV94) [reply .github/workflows/ci.yml:10]: Yes this is needed for the CI

File: pyproject.toml (hunk 2 above)
Comment (cetagostini) [reply pyproject.toml]: Done!

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**